stdlib/public/Concurrency/Task.cpp

```diff
@@ -1060,19 +1061,30 @@ void swift::swift_task_run_inline(OpaqueValue *result, void *closureAFP,
 
 SWIFT_CC(swift)
 AsyncTaskAndContext swift::swift_task_create(
-    size_t taskCreateFlags,
+    size_t rawTaskCreateFlags,
     TaskOptionRecord *options,
     const Metadata *futureResultType,
     void *closureEntry, HeapObject *closureContext) {
   FutureAsyncSignature::FunctionType *taskEntry;
   size_t initialContextSize;
-  std::tie(taskEntry, initialContextSize) =
-      getAsyncClosureEntryPointAndContextSize<
-          FutureAsyncSignature,
-          SpecialPointerAuthDiscriminators::AsyncFutureFunction>(closureEntry);
+  TaskCreateFlags taskCreateFlags(rawTaskCreateFlags);
+
+  if (taskCreateFlags.isDiscardingTask()) {
+    std::tie(taskEntry, initialContextSize) =
+        getAsyncClosureEntryPointAndContextSize<
+        FutureAsyncSignature,
+        SpecialPointerAuthDiscriminators::AsyncFutureFunction>(closureEntry);
+//            VoidAsyncSignature,
+//            SpecialPointerAuthDiscriminators::AsyncVoidFunction>(closureEntry);
```

Comment:
Could not figure out yet how to get using these Void ones to compile...

Using them causes:
```
Assertion failed: (I < getNumParams() && "parameter index out of range"), function getExtParameterInfo, file Type.h, line 4571.
PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script.
Stack dump:
0.	Program arguments: /Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/./bin/clang++ -x c++ -c /Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency/Task.cpp -o stdlib/public/Concurrency/CMakeFiles/swift_Concurrency-macosx-arm64.dir/Task.cpp.o -I/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/stdlib/public/Concurrency -I/Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency -I/Users/ktoso/code/swift-project/swift/stdlib/include -I/Users/ktoso/code/swift-project/swift/stdlib/public/SwiftShims -I/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/include -I/Users/ktoso/code/swift-project/swift/include -I/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/include -I/Users/ktoso/code/swift-project/llvm-project/clang/include -I/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/llvm-macosx-arm64/tools/clang/include -I/Users/ktoso/code/swift-project/cmark/src/include -I/Users/ktoso/code/swift-project/build/Ninja-RelWithDebInfoAssert/cmark-macosx-arm64/src -isysroot /Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -I/Users/ktoso/code/swift-project/swift/stdlib/include -isysroot /Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk -F/Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/../../../Developer/Library/Frameworks -DCMARK_STATIC_DEFINE -DGTEST_HAS_RTTI=0 -DLLVM_DISABLE_ABI_BREAKING_CHECKS_ENFORCING=1 -DSWIFT_INLINE_NAMESPACE=__runtime -D_DEBUG -D_GLIBCXX_ASSERTIONS -D_LIBCPP_ENABLE_ASSERTIONS -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -Dswift_Concurrency_macosx_arm64_EXPORTS -Wno-unknown-warning-option -Werror=unguarded-availability-new -fno-stack-protector -fPIC -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -Wimplicit-fallthrough -Wcovered-switch-default -Wno-class-memaccess -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -Werror=switch -Wdocumentation -Wimplicit-fallthrough -Wunreachable-code -Woverloaded-virtual -DOBJC_OLD_DISPATCH_PROTOTYPES=0 -O2 -DNDEBUG -arch arm64 -fPIC -fvisibility=hidden -fvisibility-inlines-hidden -Werror=gnu -Werror=c++98-compat-extra-semi -UNDEBUG -fno-sanitize=all -Dswift_Concurrency_EXPORTS -fswift-async-fp=always -D__STDC_WANT_LIB_EXT1__=1 -DSWIFT_TARGET_LIBRARY_NAME=swift_Concurrency -DSWIFT_RUNTIME -DSWIFT_LIB_SUBDIR=\"macosx\" -DSWIFT_ARCH=\"arm64\" -target arm64-apple-macosx10.13 -O2 -g -DSWIFT_ENABLE_RUNTIME_FUNCTION_COUNTERS -DSWIFT_LIBRARY_EVOLUTION=1 -DSWIFT_STDLIB_SUPPORT_BACK_DEPLOYMENT -DSWIFT_ENABLE_REFLECTION -DSWIFT_STDLIB_HAS_DLADDR -DSWIFT_STDLIB_HAS_DARWIN_LIBMALLOC=1 -DSWIFT_STDLIB_HAS_ASL -DSWIFT_STDLIB_HAS_STDIN -DSWIFT_STDLIB_HAS_ENVIRON -DSWIFT_STDLIB_HAS_LOCALE -DSWIFT_THREADING_DARWIN -DSWIFT_RUNTIME_OS_VERSIONING -DSWIFT_STDLIB_SHORT_MANGLING_LOOKUPS -DSWIFT_STDLIB_ENABLE_VECTOR_TYPES -DSWIFT_STDLIB_HAS_TYPE_PRINTING -DSWIFT_STDLIB_SUPPORTS_BACKTRACE_REPORTING -DSWIFT_STDLIB_ENABLE_UNICODE_DATA -DSWIFT_STDLIB_CONCURRENCY_TRACING -std=c++17 -fno-exceptions -fno-rtti -fno-exceptions -fno-rtti
1.	/Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency/Task.cpp:1078:79: current parser token ';'
2.	/Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency/Task.cpp:1067:53: parsing function body 'swift::swift_task_create'
3.	/Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency/Task.cpp:1067:53: in compound statement ('{}')
4.	/Users/ktoso/code/swift-project/swift/stdlib/public/Concurrency/Task.cpp:1072:43: in compound statement ('{}')
5.	/Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/tuple:1182:12: instantiating class definition 'std::tuple<void (*&)(__attribute__((swift_async_context)) swift::AsyncContext *, void *)__attribute__((swiftasynccall)), unsigned long &>::_EnableAssignFromPair<false, const std::pair<void (*)(__attribute__((swift_async_context)) swift::AsyncContext *)__attribute__((swiftasynccall)), unsigned long> &>'
6.	/Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/tuple:1172:12: instantiating class definition 'std::tuple<void (*&)(__attribute__((swift_async_context)) swift::AsyncContext *, void *)__attribute__((swiftasynccall)), unsigned long &>::_AssignPredicateFromPair<std::is_assignable, false, const std::pair<void (*)(__attribute__((swift_async_context)) swift::AsyncContext *)__attribute__((swiftasynccall)), unsigned long> &>'
7.	/Applications/XcodeSEIN.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX13.3.sdk/usr/include/c++/v1/__type_traits/is_assignable.h:26:29: instantiating class definition 'std::is_assignable<void (*&)(__attribute__((swift_async_context)) swift::AsyncContext *, void *)__attribute__((swiftasynccall)), void (*const &)(__attribute__((swift_async_context)) swift::AsyncContext *)__attribute__((swiftasynccall))>'
Stack dump without symbol names (ensure you have llvm-symbolizer in your PATH or set the environment var `LLVM_SYMBOLIZER_PATH` to point to it):
0  clang-13                 0x0000000105d7147c llvm::sys::PrintStackTrace(llvm::raw_ostream&, int) + 56
1  clang-13                 0x0000000105d7074c llvm::sys::RunSignalHandlers() + 112
2  clang-13                 0x0000000105d70c6c llvm::sys::CleanupOnSignal(unsigned long) + 236
3  clang-13                 0x0000000105ca9cac (anonymous namespace)::CrashRecoveryContextImpl::HandleCrash(int, unsigned long) + 100
4  clang-13                 0x0000000105ca9f38 CrashRecoverySignalHandler(int) + 124
5  libsystem_platform.dylib 0x00000001a03b6a84 _sigtramp + 56
6  libsystem_pthread.dylib  0x00000001a0387c28 pthread_kill + 288
7  libsystem_c.dylib        0x00000001a0295ae8 abort + 180
8  libsystem_c.dylib        0x00000001a0294e44 err + 0
9  clang-13                 0x00000001095ac5ec clang::ASTContext::mergeExtParameterInfo(clang::FunctionProtoType const*, clang::FunctionProtoType const*, bool&, bool&, llvm::SmallVectorImpl<clang::FunctionType::ExtParameterInfo>&) (.cold.4) + 0
10 clang-13                 0x0000000107ec8240 clang::ASTContext::mergeExtParameterInfo(clang::FunctionProtoType const*, clang::FunctionProtoType const*, bool&, bool&, llvm::SmallVectorImpl<clang::FunctionType::ExtParameterInfo>&) + 556
11 clang-13                 0x0000000107b68a5c clang::Sema::IsFunctionConversion(clang::QualType, clang::QualType, clang::QualType&) + 876
12 clang-13                 0x0000000107b8bd7c IsStandardConversion(clang::Sema&, clang::Expr*, clang::QualType, bool, clang::StandardConversionSequence&, bool, bool) + 1336
13 clang-13                 0x0000000107b6813c TryImplicitConversion(clang::Sema&, clang::Expr*, clang::QualType, bool, clang::Sema::AllowedExplicit, bool, bool, bool, bool) + 104
14 clang-13                 0x0000000107b6854c clang::Sema::PerformImplicitConversion(clang::Expr*, clang::QualType, clang::Sema::AssignmentAction, bool) + 272
15 clang-13                 0x00000001079123b4 clang::Sema::CheckSingleAssignmentConstraints(clang::QualType, clang::ActionResult<clang::Expr*, true>&, bool, bool, bool) + 904
16 clang-13                 0x000000010791cea4 clang::Sema::CheckAssignmentOperands(clang::Expr*, clang::ActionResult<clang::Expr*, true>&, clang::SourceLocation, clang::QualType, clang::BinaryOperatorKind) + 428
17 clang-13                 0x0000000107904f9c clang::Sema::CreateBuiltinBinOp(clang::SourceLocation, clang::BinaryOperatorKind, clang::Expr*, clang::Expr*) + 2264
```